docs/src/DeveloperDocumentation/new_developers.md

```diff
@@ -4,67 +4,74 @@ This document is meant to get new developers started. It will not go into depth
-    notifications.
+!!! note
+    Please find the technical details of the development process in the Oscar
+    [Way of Working](@ref).
 
 ## Oscar - what is the idea
 
-Oscar is the innovative, next generation Computer Algebra System. The ultimate goal for Oscar is to compete with (and ideally beat) Magma and Sage in our areas of expertise. Oscar should be accessable, even for the youngest student who is familiar with these objects. Oscar should follow general mathematical conventions to support the widest possible range of applications.
+Oscar is the innovative, next generation Computer Algebra System. The ultimate
+goal for Oscar is to compete with (and ideally beat) Magma and Sage in our
+areas of expertise. Oscar should be accessible, even for the youngest student
+who is familiar with these objects. Oscar should follow general mathematical
+conventions to support the widest possible range of applications.
 
 ## Oscar and Julia
 
 Oscar is written in Julia, but is not Julia, nor can it be. 
- Some examples to illustrate what that means:
-  Julias matrices are (C) arrays (of arbitrary dimension), parametrized by
-  the type of the entries (apart from banded, sparse, ... special matrices).
-  In the numerical world, the type mostly defines the representation of an object
-  - double and variations
-  - complex
-  - BigFloat
-  - Int
-  - BigInt
-
-  In algebra, this is either not true or terribly inefficient (or impossible)
-  Take
-    Z/nZ integers modulo n, and matrices over it
+Some examples to illustrate what that means:
+Julia's matrices are (C) arrays (of arbitrary dimension), parameterized by
+the type of the entries (apart from banded, sparse, ... special matrices).
+In the numerical world, the type mostly defines the representation of an object
+- double and variations
+- complex
+- BigFloat
+- Int
+- BigInt
+
+In algebra, this is either not true or terribly inefficient (or impossible)
+Take $\mathbb{Z}/n\mathbb{Z}$ integers modulo $n$, and matrices over it
    
-   Then either
-    n is part of the type -> every function is recompiled for every n - which kills
-    all modular (Chinese remainder theorem  (CRT) based) algorithms
+Then either:
+-  $n$ is part of the type -> every function is recompiled for every $n$ -
+  which kills all modular (Chinese remainder theorem  (CRT) based)
+  algorithms
+-  $n$ is not part of the type, then it need to be elsewhere, i.e. in the
+  parent, every element, additional arguments, ...
 
-    n is not part of the type, then it need to be elsewhere, ie. in the parent,
-    every element, additional arguments, ...
+Furthermore, if $n$ is BigInt (fmpz), so no bittype, then it cannot be part of
+the type.
 
-    Furthermore, if n is BigInt (fmpz), so no bittype, then it cannot be part of the type.
+For non-empty matrices, this does not matter as the entries have enough
+information, but for empty matrices this information needs to be collected
+elsewhere.
 
```

Comment:
The above makes it sound as if the only reason we don't use Julia's `Matrix{T}` is that we need to deal with empty matrices. But of course there is more. I wonder if we should elaborate on it (BTW all of that still does not really make it clear why our matrices can't be at least subtypes of `AbstractMatrix{T}`, but that's a separate matter I guess).

Anyway here is a first suggestion:
```suggestion

Even for non-empty matrices, there are reasons why one may not wish to
use a plain `Matrix{T}`: First off, for many operations it is important
(both mathematically and for performance reasons) that all entries of
the matrix are guaranteed to be defined over the same (identical) ring;
we cannot ensure this with `Matrix{T}`.

Moreover, just like there is `BitMatrix` one may wish to have special matrix
types that e.g. represent a matrix over $\mathbb{Z}/n\mathbb{Z}$ as a
bunch of integers plus the modulus (stored only once) for efficient
storage and operations.

```